backend/Testing/Browser/Page/BasePage.cs

```diff
@@ -46,13 +46,16 @@ public async Task<T> WaitFor()
     {
         if (UrlPattern is not null)
         {
+            //assert to get a good error message
+            await Assertions.Expect(Page).ToHaveURLAsync(UrlPattern);
```

Comment:
@myieye I added this because it gives a very helpful exception in the failure case, it actually tells you the current url and what it's been for the last couple of seconds